src/plugin-segment/analytics.js

```diff
@@ -0,0 +1,13 @@
+import ExecutionEnvironment from '@docusaurus/ExecutionEnvironment';
+
+export default (function () {
+  if (!ExecutionEnvironment.canUseDOM) {
+    return null;
+  }
+
+  return {
+    onRouteUpdate() {
```

Comment:
Required for SPAs

Reply:
This looks good... just wondering what's the point of an iife (immedietely invoked function expression) here?